.github/workflows/unset-environment-new-cli.yml

```
@@ -21,7 +21,7 @@ jobs:
     strategy:
       matrix:
         include:
-        - os: ubuntu-latest
+        - os: ubuntu-20.04
           version: stable-20210809
         - os: ubuntu-latest
```

Should this be changed to ubuntu-22.04? Or maybe above should be changed to ubuntu-latest (I think I'd prefer this). In general, we should not have to specify 22.04 (using latest instead) and when the next version of ubuntu comes out, some of our tests may break, but that's a good thing since we will be aware of this and can fix proactively.

---

Could you clarify what you mean by "Or maybe above should be changed to ubuntu-latest"?

---

Sorry...that wasn't very clear. I am suggesting that everywhere there is `ubuntu-22.04` now, we should change to `ubuntu-latest`. This was when `ubuntu-latest` is changed again, we will know if anything is broken.

In particular, I was thinking here:

https://github.com/github/codeql-action/pull/1330/files/993ca05cd7741fd4e48fe6b8a3b28f102174f4b6#diff-9bec90e4ad9a179ec1c8e1266c2ddc2fc0fbd73db55720b3425cfdf667b89e14R116

and here:

https://github.com/github/codeql-action/pull/1330/files/993ca05cd7741fd4e48fe6b8a3b28f102174f4b6#diff-9bec90e4ad9a179ec1c8e1266c2ddc2fc0fbd73db55720b3425cfdf667b89e14R116

And in other places in `.github/workflows/python-deps.yml`.

---

Ah, I see what you mean now.  I think you're generally right, but my hesitancy for `python-deps.yml` specifically is that these jobs were previously testing against both `ubuntu-20.04` and `ubuntu-22.04`, and the migration of `ubuntu-latest` to `ubuntu-22.04` isn't fully rolled out.  It would be possible for `ubuntu-latest` to start pointing to `ubuntu-20.04` again before the rollout completes, which would cause us to silently lose test coverage here.  I'd be in favour to switching to `ubuntu-latest` once the rollout completes.

---

> I'd be in favour to switching to ubuntu-latest once the rollout completes.

Sure. I'm guessing you're tracking this in https://github.com/github/codeql-core/issues/2909?